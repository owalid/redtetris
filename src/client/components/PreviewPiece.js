import React from 'react';
import { TETROMINOS } from '../plugins/tetrominos';
import Cell from './Cell';
import StyledPreview from './style/styledPreview';

const PreviewPiece = ({ sheet }) => {
  const game = { game: TETROMINOS[sheet.type].shape }
  return (
    <div>
      <StyledPreview
        isOtherUser={game.isOtherUser}
        width={game.game[0].length} 
        height={game.game.length}
      >
        {game.game.map(row => 
          row.map((cell, x) => 
            <Cell key={x} type={cell} isOtherUser={game.isOtherUser}/>
        ))}
      </StyledPreview>
    </div>
  )
}

export default PreviewPiece
