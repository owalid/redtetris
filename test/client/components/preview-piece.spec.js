import React from "react";
import '@testing-library/jest-dom/extend-expect';
import Enzyme, { shallow } from "enzyme";
import PreviewPiece from '../../../src/client/components/PreviewPiece'
import Adapter from "enzyme-adapter-react-16";
import { render } from '@testing-library/react'


Enzyme.configure({ adapter: new Adapter() });
describe('PreviewPiece component', () => {

  const Wrapper = () => (
    <PreviewPiece />
  )

  it('Is exists', () => {
    const wrapper = shallow(<Wrapper />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Can mount', () => {
    const Wr = () => (
      <PreviewPiece
        sheet={ {type: 1}}
      />
    )
    const w = render(<Wr />);
    expect(w).not.toBeNull()
  })

});