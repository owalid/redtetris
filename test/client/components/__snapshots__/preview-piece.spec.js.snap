// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewPiece component Is exists 1`] = `<PreviewPiece />`;
